### ISSUE TITLE
Test

### ISSUE THREAD

**Issue by Alesster** [Alesster/NESTJS_HM4#1]

**Comment by MatheusDaros** [comment]
Hello @Alesster 
Would it be possible for you to take a little time to provide more info:

- Token smart contract code
- Ballot smart contract code
- Scripts you are using to run the operations
- Address of the contracts in the blockchain

You can pick a [simple template](https://github.com/stevemao/github-issue-templates/blob/master/simple/ISSUE_TEMPLATE.md) or any formatting you may find better, and answer here.

**Comment by Alesster** [comment]
> Hello @Alesster Would it be possible for you to take a little time to provide more info:
> 
>     * Token smart contract code
> 
>     * Ballot smart contract code
> 
>     * Scripts you are using to run the operations
> 
>     * Address of the contracts in the blockchain
> 
> 
> You can pick a [simple template](https://github.com/stevemao/github-issue-templates/blob/master/simple/ISSUE_TEMPLATE.md) or any formatting you may find better, and answer here.

I answer here. 

I use Token smart contract, Ballot smart contract and Script to run the operations from this repository:
https://github.com/Alesster/HomeWork3_TokenizedBallot.git

Address of the contracts:
Token smart contract: 0x8D0E9A88def307e8412562E91bc44429C83cA611
Ballot smart contract: 0xf0b2a3808cd20c3dca6c82a3586962d851d79046

The code for the Nestjs is here:
https://github.com/Alesster/NESTJS_HM4.git

Is it enough information?

**Comment by MatheusDaros** [comment]
@Alesster 
Thanks for providing all this info.
Currently, the winning proposal at your ballot is indeed the proposal 0, that has `voteCount 500000000000`. Proposal 1 has 0 votes and proposal 2 has 6 votes.
So, when calling the `winningProposal` it returns `0` correctly.

Is that what you are trying to reproduce?

**Comment by Alesster** [comment]
> 

Thanks for the help. 
Yes, in this case the return is correct. The question is so huge voteCount for proposal 0. For the moment it is not clear where does it come from?

**Comment by MatheusDaros** [comment]
It is because of [this transaction](https://goerli.etherscan.io/tx/0x29a49c5f9e4b57dcbce19cf21cdf255d182eff26f3c553fe802588266bedb482)
The address `0xbdef89272d1852da9ea51a5ed130e6fed6ffe05c` voted with `500000000000` votes.

**Comment by Alesster** [comment]
then I don't understand how the votes are counted? In the script for homework3 I did vote with 500 gwei one time. Then I used the contracts in scripts of Nestjs where I voted in ethers. One vote - one ether. 
Ok, may be it is better to repeat everything from zero.

**Comment by MatheusDaros** [comment]
Yeah I noticed that in the other transactions you voted with only a few decimals. This is very common to happen actually. Each (whole) token should account for one whole vote, or 10^18 decimals of votes.

**Comment by Alesster** [comment]
Ok, now I created the new contract with 0 votes for all proposals. And in Nestjs everything works correctly. It seems it was confuse with this units coversation.
So, backend part is done. I start to realize frontend with angular. 
Thank you very much for the help.

**Comment by MatheusDaros** [comment]
Awesome 🙌🏼 
Let me know if you would need more assistance
